test/framework/filetools.py

```diff
@@ -793,6 +793,12 @@ def test_apply_patch(self):
         pattern = "I'm a toy, and very proud of it"
         self.assertTrue(pattern in patched)
 
+        toy_patch_gz = os.path.join(testdir, 'sandbox', 'sources', 'toy', 'toy-0.0_gzip.patch.gz')
+        self.assertTrue(ft.apply_patch(toy_patch_gz, path))
```

Comment:
hmm, `path` still contains the previous patched file, no?

we should clean out `self.test_prefix`, and call `extract_file` again on `toy-0.0.tar.gz` before applying the zipped patch?